### PR TITLE
Submission Ethereum Blockchain Connector – Simulated IOT Device

### PR DESCRIPTION
**Pull request by EdsonAlcala**
## Purpose

* Submission Ethereum Blockchain Connector – Simulated IOT Device

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type

What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

Read the README.md quick starts.

### CODE REVIEW

**Comment by EdsonAlcala** [comment]
@caleteeter Can you give me some feedback :) ? Thank you

**Comment by EdsonAlcala** [comment]
@mmercuri can you give me some feedback please? :)

**Comment by mmercuri** [comment]
yes - will provide feedback tonight

File: ebc-iot/README.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+public static async Task<HttpResponseMessage> Run(HttpRequest req, ILogger log)
+{
+    log.LogInformation("C# HTTP trigger function processed a request.");
+    string name = req.Query["content"];
+
+    string requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+    log.LogInformation($"Message received {requestBody}");
+    
+    string base64Decoded = ASCIIEncoding.ASCII.GetString(System.Convert.FromBase64String(requestBody));
+    log.LogInformation($"Message decoded {base64Decoded}");
```
[the remark this answers is not in the document]
Comment (EdsonAlcala) [reply ebc-iot/README.md]: yes, I have done it

File: ebc-iot/README.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+HAVING temperature > 75 AND humidity > 75
+```
+
+We have defined an average temperature and humidity range of 75.
+
+It means that, if in a window of 30 seconds, the average of temperature and humidity is greater than 75, then we will output to our Service Bus Queue a message that is going to trigger an Azure Logic App where we will execute a transaction against a real Blockchain.
+
+### Part 3: Setting up private network and Smart contract deployment
+
+Create a virtual machine -> [Here](https://portal.azure.com/#create/Canonical.UbuntuServer1604LTS-ARM)
```
[the remark this answers is not in the document]
Comment (EdsonAlcala) [reply ebc-iot/README.md]: Agreed

File: ebc-iot/README.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+> 0, 75, 0, 75
+
+Which refers to the minimum and maximum humidity and temperature.
+
+The smart contract has 1 function called IngestTelemetry, which received the humidity, temperature and timestamp.
+
+Within the function, we have a set of conditions that verify if the humidity and temperature are outside a range defined during the creation and put the state of the contract to OutOfCompliance.
+
+### Part 4: Azure logic app to trigger smart contract when an anomalie is found
```
[the remark this answers is not in the document]
Comment (EdsonAlcala) [reply ebc-iot/README.md]: thanks, I already did it

File: ebc-iot/README.md [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+
+An [IoT Hub](https://docs.microsoft.com/en-gb/azure/iot-hub/) Hub to capture data from devices.
+
+#### Azure Stream Analytics
+
+An [Azure Stream Analytics](https://docs.microsoft.com/en-gb/azure/stream-analytics/) to process data in real time
+
+#### Service Bus
+
+A [Service bus messaging service](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-quickstart-portal) to connect the output of the Azure Stream Analytics stage with a logic app that uses a Blockchain connector.
```
[the remark this answers is not in the document]
Comment (EdsonAlcala) [reply ebc-iot/README.md]: yes, done

File: ebc-iot/README.md [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+
+The IoT Hub will provide the information to the Azure Stream Analytics job and we will create a query that is going to measure the average temperature and humidity in a certain window of time.
+
+The query is located in [/extras/query.txt](./extras/query.txt)
+
+```sql
+SELECT 
+    System.Timestamp AS time,
+    Avg(temperature) AS temperature,
+    Avg(humidity) AS humidity
```
[the remark this answers is not in the document]
Comment (EdsonAlcala) [reply ebc-iot/README.md]: It just won't return anything

File: ebc-iot/README.md [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+
+Create a virtual machine -> [Here](https://portal.azure.com/#create/Canonical.UbuntuServer1604LTS-ARM)
+
+Once completed, login to your VM using SSH and [Install Docker CE Edition](https://docs.docker.com/install/linux/docker-ce/ubuntu/)
+
+Once you have finished and docker is up and running
+
+Run
+
+docker run -t -d --name quorum-n-nodes -p 22001-22004:22001-22004 edsonalcala/quorum-n-nodes:1.1-alpine
```
[the remark this answers is not in the document]
Comment (EdsonAlcala) [reply ebc-iot/README.md]: I removed this part already

File: ebc-iot/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (EdsonAlcala) [reply ebc-iot/README.md]: I did the font bigger, can you take a look?

**Comment by caleteeter** [comment]
Accepting.

**Pull request merged**